## Project

### Main Goal
* Doing market analysis for an e-learning company focused on web and mobile development
* Find the two best markets to advertise our product in using data from a survey

In [7]:
import pandas as pd
import seaborn as sns

## Explore Data

In [2]:
df = pd.read_csv('2017-fCC-New-Coders-Survey-Data.csv')

/dataquest/system/env/python3/lib/python3.4/site-packages/IPython/core/interactiveshell.py:2723: DtypeWarning: Columns (17,62) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
print(df.head())

    Age  AttendedBootcamp  BootcampFinish  BootcampLoanYesNo BootcampName  \
0  27.0               0.0             NaN                NaN          NaN   
1  34.0               0.0             NaN                NaN          NaN   
2  21.0               0.0             NaN                NaN          NaN   
3  26.0               0.0             NaN                NaN          NaN   
4  20.0               0.0             NaN                NaN          NaN   

   BootcampRecommend  ChildrenNumber                 CityPopulation  \
0                NaN             NaN            more than 1 million   
1                NaN             NaN              less than 100,000   
2                NaN             NaN            more than 1 million   
3                NaN             NaN  between 100,000 and 1 million   
4                NaN             NaN  between 100,000 and 1 million   

   CodeEventConferences  CodeEventDjangoGirls         ...           \
0                   NaN                 

In [5]:
print(df.shape)

(18175, 136)


In [6]:
print(df.isnull().sum())

Age                        2808
AttendedBootcamp            466
BootcampFinish            17106
BootcampLoanYesNo         17096
BootcampName              17226
BootcampRecommend         17102
ChildrenNumber            15861
CityPopulation             2821
CodeEventConferences      16566
CodeEventDjangoGirls      18010
CodeEventFCC              16467
CodeEventGameJam          17885
CodeEventGirlDev          17878
CodeEventHackathons       16198
CodeEventMeetup           15703
CodeEventNodeSchool       17729
CodeEventNone             17099
CodeEventOther            17605
CodeEventRailsBridge      18042
CodeEventRailsGirls       18043
CodeEventStartUpWknd      17631
CodeEventWkdBootcamps     17628
CodeEventWomenCode        17694
CodeEventWorkshops        16422
CommuteTime                8929
CountryCitizen             2793
CountryLive                2839
EmploymentField           10059
EmploymentFieldOther      17097
EmploymentStatus           3830
                          ...  
Resource

There 136 columns of data and a lot of these columns are missing values - it seems like if no response was given, no value was written to that row. Given the number of columns - the best way to approach this would be reduce the dataframe to answer the most pertinent questions:

1. Where are new coders located?
2. What are the locations with the greatest number of new coders?
3. How much money new coders are willing to spend on learning?
4. How many people are interested in learning about mobile & web dev?



## Analyzing Job Role Interest

1. Are people interested in only one subject or multiple?
2. If most people are interested in more than one subject - is the sample still representative?
3. How many people are interested in at least one of those two subjects?

In [10]:
df.JobRoleInterest.value_counts()

Full-Stack Web Developer                                                                                                                                                                                         823
  Front-End Web Developer                                                                                                                                                                                        450
  Data Scientist                                                                                                                                                                                                 152
Back-End Web Developer                                                                                                                                                                                           142
  Mobile Developer                                                                                                                                  